server/sonar-server/src/main/java/org/sonar/server/project/ws/IndexAction.java

```diff
@@ -0,0 +1,108 @@
+ */
+
+package org.sonar.server.project.ws;
+
+import com.google.common.io.Resources;
+import java.util.Optional;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.server.user.UserSession;
+
+import static java.util.Optional.ofNullable;
+import static org.sonar.server.ws.KeyExamples.KEY_PROJECT_EXAMPLE_001;
+import static org.sonarqube.ws.client.project.ProjectsWsParameters.ACTION_INDEX;
+import static org.sonarqube.ws.client.project.ProjectsWsParameters.PARAM_KEY;
+
+public class IndexAction implements ProjectsWsAction {
+
+  private static final String PARAM_SEARCH = "search";
+  private static final String PARAM_SUB_PROJECTS = "subprojects";
+  private static final String PARAM_FORMAT = "format";
+  private static final String REMOVED_PARAMETER_DESCRIPTION = "Since 6.3, this parameter has no effect";
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+
+  public IndexAction(DbClient dbClient, UserSession userSession) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION_INDEX)
+      .setDescription("This web service is deprecated, please use api/components/search instead")
+      .setSince("2.10")
+      .setDeprecatedKey("6.3")
+      .setHandler(this)
+      .setResponseExample(Resources.getResource(this.getClass(), "index-example.json"));
+    action.createParam(PARAM_KEY)
+      .setDescription("key or id of the project")
+      .setExampleValue(KEY_PROJECT_EXAMPLE_001);
+    action.createParam(PARAM_SEARCH)
+      .setDescription("Substring of project name, case insensitive. Ignored if the parameter key is set")
+      .setExampleValue("Sonar");
+    action.createParam(PARAM_SUB_PROJECTS)
+      .setDescription("Load sub-projects. Ignored if the parameter key is set")
+      .setDefaultValue("false")
+      .setBooleanPossibleValues();
+    action.createParam(PARAM_FORMAT)
+      .setDescription("Only json response format is available")
+      .setPossibleValues("json");
+    addRemovedParameter("desc", action);
+    addRemovedParameter("views", action);
+    addRemovedParameter("libs", action);
+    addRemovedParameter("versions", action);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      String projectKey = request.param(PARAM_KEY);
+      if (projectKey != null) {
+        Optional<ComponentDto> project = getProjectByKeyOrId(dbSession, projectKey);
+      } else {
+        String search = request.param(PARAM_SEARCH);
+        boolean loadModules = request.paramAsBoolean(PARAM_SUB_PROJECTS);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused "loadModules" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1481)